allennlp/data/tokenizers/pretrained_transformer_tokenizer.py

```diff
@@ -232,6 +233,9 @@ def intra_word_tokenize_in_id(
         cumulative = starting_offset
         for token in tokens:
             subword_wordpieces = self.tokenizer.encode(token, add_special_tokens=False)
+            if len(subword_wordpieces) == 0:
```

Comment:
How can this happen? When it is a whitespace character?

So I think this is what was happening with the BERT model in the demo. Because if this was returning an empty list, all the offsets into the original tokens would be incorrect (I think). 

This needs a test, and also it would be great if you could verify that this was what was going wrong with the model in the demo, because we need to be absolutely sure of what's happening here and have it very well documented in the docstring of this class.

Reply:
The most trivial case would be a whitespace character, but there could also some other unicode characters like `"\u007f"`. Agreed that this should be the cause of the demo problem. It's not that the offsets will be wrong -- they still are correct. The problem is that in the mismatched embedder, the length of these spans are 0. And when I needed the average span representation, there will be divide-by-zero's that cause Nan's which mess up all following tensor operations. So it's very likely that this is the cause, but I can verify later.

https://github.com/allenai/allennlp/blob/master/allennlp/modules/token_embedders/pretrained_transformer_mismatched_embedder.py#L82-L83